### PR TITLE
TileType class refactoring

### PR DESCRIPTION
**Pull request by Grenkin1988**
Made some refactoring for TileType

### CODE REVIEW

**Comment by bjubes** [comment]
why did u change the spacing in Tiles.xml, or was that accidental?

**Comment by Grenkin1988** [comment]
A can undo this change to Tiles.xml, but that is how file looks like after autoformat.

**Comment by bjubes** [comment]
I don't think we have a standard for amount of spaces to indent in XML, so your probably fine

File: Assets/Scripts/Models/TileType.cs [hunk 1]
```diff
@@ -6,87 +6,78 @@
...
     /// </summary>
-    public float PathfindingModifier
-    {
-        get { return pathfindingModifier; }
-        set { pathfindingModifier = value; }
-    }
+    public float PathfindingModifier { get; set; }
 
-    // TODO!
     public bool LinksToNeighbours { get; protected set; }
```
Comment (TomMalbran) [inline Assets/Scripts/Models/TileType.cs]: Should you use private here, and in the next few properties? Is changed above from protected.

File: Assets/Scripts/Models/TileType.cs [hunk 2]
```diff
@@ -207,104 +172,64 @@ public override int GetHashCode()
...
+        XmlReader reader = parentReader.ReadSubtree();
         while (reader.Read())
         {
             switch (reader.Name)
             {
                 case "Name":
-                    reader.Read();
-                    Name = reader.ReadContentAsString();
-                    break;
+                reader.Read();
```
Comment (TomMalbran) [inline Assets/Scripts/Models/TileType.cs]: I thought that we were doing swicthes like:

```
switch (reader.Name)
{
    case "Name":
        // Code goes here
        break;
    case "Description":
        // ...
        break;
}
```

It was already changed in furniture from a different style.

**Pull request merged**